fontc/src/timing.rs

```diff
@@ -206,101 +218,73 @@ fn color(id: &AnyWorkId) -> &'static str {
     }
 }
 
-/// The initial state of a runnable job.
-///
-/// It may have queued from t0 to launchable but now it's go-time!
-pub struct JobTimeRunnable {
+/// Inner state for timing tasks.
+#[derive(Debug, Clone)]
+pub struct JobTimeState {
     id: AnyWorkId,
     nth_wave: usize,
-    runnable: Instant,
-}
-
-impl JobTimeRunnable {
-    /// Time of submission to execution queue, e.g. thread pool submission
-    pub fn queued(self) -> JobTimeQueued {
-        JobTimeQueued {
-            id: self.id,
-            nth_wave: self.nth_wave,
-            runnable: self.runnable,
-            queued: Instant::now(),
-        }
-    }
+    thread_id: ThreadId,
+    queued: Instant,
+    run: Instant,
+    complete: Instant,
 }
 
-pub struct JobTimeQueued {
-    id: AnyWorkId,
-    nth_wave: usize,
-    runnable: Instant,
-    queued: Instant,
+/// A state machine tracking timer state.
+#[derive(Debug, Clone)]
+pub enum JobTime {
+    /// Initial state of a runnable job.
+    Runnable(JobTimeState),
+    /// State after submission to an execution queue.
+    Queued(JobTimeState),
+    /// State after the job starts running
+    Running(JobTimeState),
+    /// State when the job has finished.
+    Done(JobTimeState),
```

Comment:
These comments just duplicate what the type tells you. "Running(State) is the state when running" is ... noise. I suggest we instead migrate some of the old comments from each state, e.g. queued has notes about threads, etc.